tutorials/xr/xr_action_map.rst

```diff
@@ -0,0 +1,242 @@
+
+.. image:: img/xr_aim_pose.webp
+
+The columns in our table are as follows:
+
+.. list-table::
+  :class: wrap-normal
+  :width: 100%
+  :widths: 7 23 70
+  :header-rows: 1
+
+  * - Col
+    - Value
+    - Description
+  * - 1
+    - aim_pose
+    - This is the internal name of the action. OpenXR doesn't specify specific restrictions on this name other then size, however some XR runtimes will not like spaces or special characters.
+  * - 2
+    - Aim pose
+    - This is a human-readable name for the action. Some XR runtimes will display this name to the end user, for example in configuration dialogs.
+  * - 3
+    - Pose
+    - The type of this action.
+
+OpenXR defines a number of input poses that are commonly available for controllers and can thus be bound however there is no rule which poses are supported for different controllers. The poses OpenXR currently defines are:
+
+  * The aim pose on most controllers is positioned slightly infront of the controller and aims forward. This is a great pose to use for laser pointers or to align the muzzle of a weapon with.
+  * The grip pose on most controllers is positioned where the grip button is placed on the controller. The orientation of this pose differs between controllers and can differ for the same controller on different XR runtimes.
+  * The palm pose on most controllers is positioned in the center of the palm of the hand holding the controller. This is a new pose that is not available on all XR runtimes.
+
+ .. note::
+  If hand tracking is used there are currently big differences in implementations between the different XR runtimes. As a result the action map is currently not suitable for hand tracking. Work is being done on this so stay tuned.
+
+
+Lets complete our list of actions for a very simple shooting game/application:
+
+.. image:: img/xr_all_actions.webp
+
+The actions we have added are:
+
+  * movement, which allows the user to move around outside of normal room scale tracking.
+  * grab, which detects the user wants to hold something.
+  * shoot, which detects the user wants to fire the weapon they are holding.
+  * haptic, which allows us to output haptic feedback.
+
+Now note that we don't make a difference between the left and right hand. This is something that is determined at the next stage. We've implemented the action system in such a way that you can bind the same action to both hands, and the correct :ref:`XRController3D <class_xrcontroller3d>` node will emit the signal.
+
+.. warning::
+  Note that for both grab and shoot we've used the ``Bool`` type. As mentioned before, OpenXR does automatic conversions from an analogue controls however not all XR Runtimes currently apply sensible thresholds.
+
+  We recommend as a workaround to use the ``Float`` type when interacting with triggers and grip buttons and apply your own threshold.
+
+  For A/B/X/Y buttons the ``Bool`` type works fine.
```

Comment:
This may be confusing since not all controllers have the same button layouts. I'm not sure if it adds to the `warning` so it could be removed.

Reply:
I've changed the text here, I also added a note giving some more clarity around click and touch variants of inputs.